When Kueue panics during integration tests with INTEGRATION_NPROC>1 the panic is not logged

**What happened**:

We investigated issue with branch failing https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/5287/pull-kueue-test-integration-baseline-main/1925602953855504384

as follows:

```
  In [ReportAfterSuite] at: autogenerated by Ginkgo @ 05/22/25 17:26:39.113
------------------------------
Ginkgo timed out waiting for all parallel procs to report back
Test suite: tas (./test/integration/singlecluster/tas)
This occurs if a parallel process exits before it reports its results to the
Ginkgo CLI.  The CLI will now print out all the stdout/stderr output it's
collected from the running processes.  However you may not see anything useful
in these logs because the individual test processes usually intercept output to
stdout/stderr in order to capture it in the spec reports.
You may want to try rerunning your test suite with
--output-interceptor-mode=none to see additional output here and debug your
suite.
  
Output from proc 1:
  --- FAIL: TestAPIs (63.54s)
  FAIL
Output from proc 2:
  PASS
Output from proc 3:
  PASS
Output from proc 4:
** End **Could not open /logs/artifacts/test_integration_singlecluster_tas_integration.json:
open /logs/artifacts/test_integration_singlecluster_tas_integration.json: no such file or directory
Could not open /logs/artifacts/test_integration_singlecluster_tas_junit.xml:
open /logs/artifacts/test_integration_singlecluster_tas_junit.xml: no such file or directory
Ginkgo ran 1 suite in 2m47.357265802s
```
However, when we run the same branch locally with INTEGRATION_NPROC=1, then we could see the panic:

```
00,"pods":1}}
E0522 19:29:39.312804  533523 panic.go:262] "Observed a panic" panic="runtime error: invalid memory address or nil pointer dereference" panicGoValue="\"invalid memory address or nil pointer dereference\"" stacktrace=<
    goroutine 410 [running]:
    [k8s.io/apimachinery/pkg/util/runtime.logPanic({0x3438b60](http://k8s.io/apimachinery/pkg/util/runtime.logPanic%28%7B0x3438b60), 0x46d7880}, {0x2d40b80, 0x465eda0})
        <HOME>go/src/[sigs.k8s.io/kueue/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:107](http://sigs.k8s.io/kueue/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:107) +0xdd
    [k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x3438b60](http://k8s.io/apimachinery/pkg/util/runtime.handleCrash%28%7B0x3438b60), 0x46d7880}, {0x2d40b80, 0x465eda0}, {0x46d7880, 0x0, 0xc001069a88?})
        <HOME>go/src/[sigs.k8s.io/kueue/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:82](http://sigs.k8s.io/kueue/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:82) +0x9a
    [k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0](http://k8s.io/apimachinery/pkg/util/runtime.HandleCrash%28%7B0x0), 0x0, 0x3147e70?})
        <HOME>go/src/[sigs.k8s.io/kueue/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:59](http://sigs.k8s.io/kueue/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:59) +0x18c
    panic({0x2d40b80?, 0x465eda0?})
        <HOME>sdk/go1.24.0/src/runtime/panic.go:787 +0x132
    [sigs.k8s.io/kueue/pkg/cache.(*TASFlavorSnapshot).mergeTopologyAssignments.func1(0x1](http://sigs.k8s.io/kueue/pkg/cache.%28*TASFlavorSnapshot%29.mergeTopologyAssignments.func1%280x1), 0x0)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/cache/tas_flavor_snapshot.go:560](http://sigs.k8s.io/kueue/pkg/cache/tas_flavor_snapshot.go:560) +0x290
    sort.insertionSort_func({0xc001069ec0?, 0xc0016944b0?}, 0x0, 0x2)
        <HOME>sdk/go1.24.0/src/sort/zsortfunc.go:12 +0xc7
    sort.pdqsort_func({0xc001069ec0?, 0xc0016944b0?}, 0x0, 0x2, 0x2)
        <HOME>sdk/go1.24.0/src/sort/zsortfunc.go:73 +0x3a5
    sort.Slice({0x2c17f60, 0xc0009911a0}, 0xc001069ec0)
        <HOME>sdk/go1.24.0/src/sort/slice.go:29 +0xab
    [sigs.k8s.io/kueue/pkg/cache.(*TASFlavorSnapshot).mergeTopologyAssignments(0xc00169ddd0](http://sigs.k8s.io/kueue/pkg/cache.%28*TASFlavorSnapshot%29.mergeTopologyAssignments%280xc00169ddd0), 0xc001694480, 0xc001632720)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/cache/tas_flavor_snapshot.go:557](http://sigs.k8s.io/kueue/pkg/cache/tas_flavor_snapshot.go:557) +0x318
    [sigs.k8s.io/kueue/pkg/cache.(*TASFlavorSnapshot).FindTopologyAssignmentsForFlavor(0xc00169ddd0](http://sigs.k8s.io/kueue/pkg/cache.%28*TASFlavorSnapshot%29.FindTopologyAssignmentsForFlavor%280xc00169ddd0), {0xc000341630, 0x1, 0xa?}, 0x0, 0xc0004e4800)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/cache/tas_flavor_snapshot.go:443](http://sigs.k8s.io/kueue/pkg/cache/tas_flavor_snapshot.go:443) +0x6b6
    [sigs.k8s.io/kueue/pkg/cache.(*ClusterQueueSnapshot).FindTopologyAssignmentsForWorkload(0xc00192aa20](http://sigs.k8s.io/kueue/pkg/cache.%28*ClusterQueueSnapshot%29.FindTopologyAssignmentsForWorkload%280xc00192aa20), 0xc001694330, 0x0, 0xc0004e4800)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/cache/clusterqueue_snapshot.go:217](http://sigs.k8s.io/kueue/pkg/cache/clusterqueue_snapshot.go:217) +0x1c5
    [sigs.k8s.io/kueue/pkg/scheduler/flavorassigner.(*FlavorAssigner).assignFlavors(0xc00106ad68](http://sigs.k8s.io/kueue/pkg/scheduler/flavorassigner.%28*FlavorAssigner%29.assignFlavors%280xc00106ad68), {{0x343f668?, 0xc001694090?}, 0x46d7880?}, {0x0, 0x0, 0x18?})
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/scheduler/flavorassigner/flavorassigner.go:469](http://sigs.k8s.io/kueue/pkg/scheduler/flavorassigner/flavorassigner.go:469) +0xe7a
    [sigs.k8s.io/kueue/pkg/scheduler/flavorassigner.(*FlavorAssigner).Assign(0xc00106ad68](http://sigs.k8s.io/kueue/pkg/scheduler/flavorassigner.%28*FlavorAssigner%29.Assign%280xc00106ad68), {{0x343f668?, 0xc001694090?}, 0x4c1629?}, {0x0, 0x0, 0x0})
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/scheduler/flavorassigner/flavorassigner.go:389](http://sigs.k8s.io/kueue/pkg/scheduler/flavorassigner/flavorassigner.go:389) +0x4b6
    [sigs.k8s.io/kueue/pkg/scheduler.(*Scheduler).getInitialAssignments(0xc0002c1c20](http://sigs.k8s.io/kueue/pkg/scheduler.%28*Scheduler%29.getInitialAssignments%280xc0002c1c20), {{0x343f668?, 0xc001694090?}, 0xc001957a70?}, 0xc0006a6540, 0xc001387500)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:455](http://sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:455) +0x354
    [sigs.k8s.io/kueue/pkg/scheduler.(*Scheduler).getAssignments(0xc0002c1c20](http://sigs.k8s.io/kueue/pkg/scheduler.%28*Scheduler%29.getAssignments%280xc0002c1c20), {{0x343f668?, 0xc001694090?}, 0x1?}, 0xc0006a6540, 0xc001387500)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:446](http://sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:446) +0xf0
    [sigs.k8s.io/kueue/pkg/scheduler.(*Scheduler).nominate(0xc0002c1c20](http://sigs.k8s.io/kueue/pkg/scheduler.%28*Scheduler%29.nominate%280xc0002c1c20), {0x3438cd0, 0xc001957a70}, {0xc000ae6dc0, 0x1, 0x0?}, 0xc001387500)
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:380](http://sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:380) +0x1205
    [sigs.k8s.io/kueue/pkg/scheduler.(*Scheduler).schedule(0xc0002c1c20](http://sigs.k8s.io/kueue/pkg/scheduler.%28*Scheduler%29.schedule%280xc0002c1c20), {0x3438cd0, 0xc000afc930})
        <HOME>go/src/[sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:199](http://sigs.k8s.io/kueue/pkg/scheduler/scheduler.go:199) +0x40e
    [sigs.k8s.io/kueue/pkg/util/wait.untilWithBackoff.func1()](http://sigs.k8s.io/kueue/pkg/util/wait.untilWithBackoff.func1%28%29)
```

**What you expected to happen**:

The panic is always put into the build artifacts.

**How to reproduce it (as minimally and precisely as possible)**:

Run integration tests, but inject into Kueue code panicing.

**Anything else we need to know?**:

It is probably the same here https://github.com/kubernetes-sigs/kueue/issues/5304, which makes the rare flake very hard to investigate. Also, the fact that we don't see it panics makes people less concerned about the flake like that.

--- a/Makefile-test.mk
+++ b/Makefile-test.mk
@@ -34,6 +34,7 @@ INTEGRATION_TARGET_MULTIKUEUE ?= ./test/integration/multikueue/...
 # Verbosity level for apiserver logging.
 # The logging is disabled if 0.
 INTEGRATION_API_LOG_LEVEL ?= 0
+INTEGRATION_OUTPUT_OPTIONS ?= --output-interceptor-mode=none 
 
 # Folder where the e2e tests are located.
 E2E_TARGET ?= ./test/e2e/...
@@ -85,7 +86,7 @@ test-integration: gomod-download envtest ginkgo dep-crds kueuectl ginkgo-top ## 
 	KUEUE_BIN=$(PROJECT_DIR)/bin \
 	ENVTEST_K8S_VERSION=$(ENVTEST_K8S_VERSION) \
 	TEST_LOG_LEVEL=$(TEST_LOG_LEVEL) API_LOG_LEVEL=$(INTEGRATION_API_LOG_LEVEL) \
-	$(GINKGO) $(INTEGRATION_FILTERS) $(GINKGO_ARGS) $(GOFLAGS) -procs=$(INTEGRATION_NPROCS) --race --junit-report=junit.xml --json-report=integration.json --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
+	$(GINKGO) $(INTEGRATION_FILTERS) $(GINKGO_ARGS) $(GOFLAGS) -procs=$(INTEGRATION_NPROCS) --race --junit-report=junit.xml --json-report=integration.json $(INTEGRATION_OUTPUT_OPTIONS) --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
 	$(PROJECT_DIR)/bin/ginkgo-top -i $(ARTIFACTS)/integration.json > $(ARTIFACTS)/integration-top.yaml
 
 .PHONY: test-integration-baseline
@@ -103,7 +104,7 @@ test-multikueue-integration: gomod-download envtest ginkgo dep-crds ginkgo-top #
 	KUEUE_BIN=$(PROJECT_DIR)/bin \
 	ENVTEST_K8S_VERSION=$(ENVTEST_K8S_VERSION) \
 	TEST_LOG_LEVEL=$(TEST_LOG_LEVEL) API_LOG_LEVEL=$(INTEGRATION_API_LOG_LEVEL) \
-	$(GINKGO) $(INTEGRATION_FILTERS) $(GINKGO_ARGS) $(GOFLAGS) -procs=$(INTEGRATION_NPROCS_MULTIKUEUE) --race --junit-report=multikueue-junit.xml --json-report=multikueue-integration.json --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET_MULTIKUEUE)
+	$(GINKGO) $(INTEGRATION_FILTERS) $(GINKGO_ARGS) $(GOFLAGS) -procs=$(INTEGRATION_NPROCS_MULTIKUEUE) --race --junit-report=multikueue-junit.xml --json-report=multikueue-integration.json $(INTEGRATION_OUTPUT_OPTIONS) --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET_MULTIKUEUE)
 	$(PROJECT_DIR)/bin/ginkgo-top -i $(ARTIFACTS)/multikueue-integration.json > $(ARTIFACTS)/multikueue-integration-top.yaml
 
 CREATE_KIND_CLUSTER ?= true

